packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,364 @@
+            } else {
+                const cookieConsent = this.$cookies.get('je-cookieConsent');
+                this.hideBanner = cookieConsent === 'full' || cookieConsent === 'necessary';
+            }
+        },
+        /**
+         * Set the cookie for the user's choice
+         * @param {String} cookieValue
+         */
+        setCookieBannerCookie (cookieValue) {
+            this.$cookies.set('je-cookieConsent', cookieValue, {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Set the legacy cookie banner cookie
+         */
+        setLegacyCookieBannerCookie () {
+            this.$cookies.set('je-banner_cookie', '2', {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Push tracking events
+         * @param {String} consentLevel
+         */
+        dataLayerPush (consentLevel) {
+            const dataLayer = window.dataLayer || [];
+            dataLayer.push({ event: 'trackConsent', userData: { consent: consentLevel } });
+        },
+        /**
+         * Check for excluded cookies/storage
+         * @param {String} cookieName
+         * @returns {Bool}
+         */
+        isNotExcluded (cookieName) {
+            return this.config.cookieExclusionList.every(arrVal => cookieName.lastIndexOf(arrVal, arrVal.length - 1) === -1);
+        },
+        /**
+         * Remove unnecessary cookies
+         */
+        removeUnnecessaryCookies () {
+            const cookies = Object.keys(this.$cookies.getAll());
+            for (let i = 0; i < cookies.length; i++) {
+                if (this.isNotExcluded(cookies[i])) this.$cookies.remove(cookies[i]);
+            }
+        },
+        /**
+         * Check for iOS browser
+         * @returns {Bool}
+         */
+        checkIosSafari () {
+            return /(iPhone|iPad).*Safari/.test(navigator.userAgent);
+        },
+        /**
+         * Resend GTM events
+         */
+        resendEvents () {
+            const dataLayer = window.dataLayer || [];
+            const excludeEvents = ['gtm.js', 'gtm.click', 'gtm.linkClick', 'gtm.triggerGroup', 'trackConsent', 'showConsent'];
+            let i;
+            for (i = 0; i < dataLayer.length; i++) {
+                if (dataLayer[i].event && !excludeEvents.includes(dataLayer[i].event)) {
+                    dataLayer[i].eventResent = dataLayer[i].event;
+                    dataLayer[i]['gtm.uniqueEventId'] = Math.floor(100000 + Math.random() * 900000);
+                    dataLayer.push(dataLayer[i]);
+                }
+                if (dataLayer[i].event && dataLayer[i].event === 'trackConsent' && (dataLayer[i].userData.consent === 'full' || dataLayer[i].userData.consent === 'necessary')) {
+                    dataLayer.push({ eventResent: false });
+                    break;
+                }
+            }
+        }
```

Comment:
I'm not sure if we got this from somewhere else, but it looks rather complex and hard to understand. Is it something we can simplify?

Reply:
@SachaZvetelman This is from GTM, so I'll ask Pete Crawley/Rob Ellison if they can provide a simplified version

Reply:
@SachaZvetelman Pete reckons this is as simple as it can be. Besides, he thinks we should keep third party code as is, in case we need to get Pete/Rob involved in debugging, so that they'll recognise and understand the code. As this is GTM code and not our code, I agree with Pete. If you really really think we need to refactor this code, then I think we'll need to carefully document any changes to help the data guys understand what we've done and why.